diff -dur psmisc-22.21.orig/src/fuser.c psmisc-22.21/src/fuser.c
--- psmisc-22.21.orig/src/fuser.c	2014-01-31 14:07:32.000000000 +0100
+++ psmisc-22.21/src/fuser.c	2016-04-22 14:59:25.000000000 +0200
@@ -1871,7 +1871,7 @@
 				len = write(pipes[1], buf, sizeof(struct stat));
 			while (len < 0 && errno == EINTR);
 		close(pipes[1]);
-		exit(ret);
+		_exit(ret);
 	default:
 		close(pipes[1]);
 		if (sigsetjmp(jenv, 1)) {
