--- psmisc-21.3/configure.in.orig	2003-07-11 08:00:55.000000000 +0200
+++ psmisc-21.3/configure.in	2003-07-11 08:10:30.000000000 +0200
@@ -19,9 +19,10 @@
 AC_SUBST(FLASK_LIB)
 
 dnl Checks for libraries.
+AC_CHECK_LIB(tinfo, tgetent, TERMCAP_LIB=-ltinfo,
 AC_CHECK_LIB(ncurses, tgetent, TERMCAP_LIB=-lncurses,
   AC_CHECK_LIB(termcap, tgetent, TERMCAP_LIB=-ltermcap, 
-    AC_MSG_ERROR(Cannot find ncurses or termcap libraries) ) )
+    AC_MSG_ERROR(Cannot find ncurses or termcap libraries) ) ) )
 AC_SUBST(TERMCAP_LIB)
 
 dnl Checks for header files.
