--- psmisc-21.5/src/killall.c.orig	2004-04-23 15:51:26.000000000 +0200
+++ psmisc-21.5/src/killall.c	2004-07-16 00:44:55.655536344 +0200
@@ -382,13 +382,13 @@
 static void
 usage_pidof (void)
 {
-  fprintf (stderr, "usage: pidof [ -eg ] name ...\n");
-  fprintf (stderr, "       pidof -V\n\n");
-  fprintf (stderr, "    -e      require exact match for very long names;\n");
-  fprintf (stderr, "            skip if the command line is unavailable\n");
+  fprintf (stderr, _("usage: pidof [ -eg ] name ...\n"));
+  fprintf (stderr, _("       pidof -V\n\n"));
+  fprintf (stderr, _("    -e      require exact match for very long names;\n"));
+  fprintf (stderr, _("            skip if the command line is unavailable\n"));
   fprintf (stderr,
-	   "    -g      show process group ID instead of process ID\n");
-  fprintf (stderr, "    -V      display version information\n\n");
+	   _("    -g      show process group ID instead of process ID\n"));
+  fprintf (stderr, _("    -V      display version information\n\n"));
 }
 
 
@@ -396,26 +396,26 @@
 usage_killall (void)
 {
 #ifdef FLASK_LINUX
-  fprintf(stderr,"Usage: killall [-s sid] [-c context] [ -egiqvw ] [ -signal ] name ...\n");
+  fprintf(stderr,_("Usage: killall [-s sid] [-c context] [ -egiqvw ] [ -signal ] name ...\n"));
 #else  /*FLASK_LINUX*/
-  fprintf (stderr, "usage: killall [ OPTIONS ] [ -- ] name ...\n");
+  fprintf (stderr, _("usage: killall [ OPTIONS ] [ -- ] name ...\n"));
 #endif /*FLASK_LINUX*/
-  fprintf (stderr, "       killall -l, --list\n");
-  fprintf (stderr, "       killall -V --version\n\n");
-  fprintf (stderr, "  -e,--exact          require exact match for very long names\n");
-  fprintf (stderr, "  -I,--ignore-case-   case insensitive process name match\n");
-  fprintf (stderr, "  -g,--process-group  kill process group instead of process\n");
-  fprintf (stderr, "  -i,--interactive    ask for confirmation before killing\n");
-  fprintf (stderr, "  -l,--list           list all known signal names\n");
-  fprintf (stderr, "  -q,--quiet          don't print complaints\n");
-  fprintf (stderr, "  -s,--signal         send signal instead of SIGTERM\n");
-  fprintf (stderr, "  -v,--verbose        report if the signal was successfully sent\n");
-  fprintf (stderr, "  -V,--version        display version information\n");
-  fprintf (stderr, "  -w,--wait           wait for processes to die\n\n");
-#ifdef FLASK_LINUX
-  fprintf (stderr, "  -d,--sid            kill only process(es) having sid\n");
-  fprintf (stderr, "  -c,--context        kill only process(es) having scontext\n");
-  fprintf(stderr, "   (-s, -c are mutually exclusive and must precede other arguments)\n\n");
+  fprintf (stderr, _("       killall -l, --list\n"));
+  fprintf (stderr, _("       killall -V --version\n\n"));
+  fprintf (stderr, _("  -e,--exact          require exact match for very long names\n"));
+  fprintf (stderr, _("  -I,--ignore-case-   case insensitive process name match\n"));
+  fprintf (stderr, _("  -g,--process-group  kill process group instead of process\n"));
+  fprintf (stderr, _("  -i,--interactive    ask for confirmation before killing\n"));
+  fprintf (stderr, _("  -l,--list           list all known signal names\n"));
+  fprintf (stderr, _("  -q,--quiet          don't print complaints\n"));
+  fprintf (stderr, _("  -s,--signal         send signal instead of SIGTERM\n"));
+  fprintf (stderr, _("  -v,--verbose        report if the signal was successfully sent\n"));
+  fprintf (stderr, _("  -V,--version        display version information\n"));
+  fprintf (stderr, _("  -w,--wait           wait for processes to die\n\n"));
+#ifdef FLASK_LINUX
+  fprintf (stderr, _("  -d,--sid            kill only process(es) having sid\n"));
+  fprintf (stderr, _("  -c,--context        kill only process(es) having scontext\n"));
+  fprintf(stderr, _("   (-s, -c are mutually exclusive and must precede other arguments)\n\n"));
 #endif /*FLASK_LINUX*/
 }
 
@@ -433,11 +433,11 @@
 void print_version()
 {
   fprintf(stderr, "%s (psmisc) %s\n", pidof ? "pidof" : "killall", VERSION);
-  fprintf(stderr, "Copyright (C) 1993-2002 Werner Almesberger and Craig Small\n\n");
-  fprintf(stderr, "PSmisc comes with ABSOLUTELY NO WARRANTY.\n");
-  fprintf(stderr, "This is free software, and you are welcome to redistribute it under the terms\n");
-  fprintf(stderr, "of the GNU General Public License.\n");
-  fprintf(stderr, "For more information about these matters, see the files named COPYING.\n");
+  fprintf(stderr, _("Copyright (C) 1993-2002 Werner Almesberger and Craig Small\n\n"));
+  fprintf(stderr, _("PSmisc comes with ABSOLUTELY NO WARRANTY.\n"));
+  fprintf(stderr, _("This is free software, and you are welcome to redistribute it under the terms\n"));
+  fprintf(stderr, _("of the GNU General Public License.\n"));
+  fprintf(stderr, _("For more information about these matters, see the files named COPYING.\n"));
 }
 
 int
@@ -551,7 +551,7 @@
 
 	  lsid = strtol(optarg, buf, 0);
           if ( **buf ) {
-              (void) fprintf(stderr, "%s: SID (%s) must be numeric\n", name, *argv);
+              (void) fprintf(stderr, _("%s: SID (%s) must be numeric\n"), name, *argv);
               (void) fflush(stderr);
               return( 1 );
           }
--- psmisc-21.5/src/fuser.c.orig	2004-04-30 07:25:09.000000000 +0200
+++ psmisc-21.5/src/fuser.c	2004-07-16 00:46:11.442015048 +0200
@@ -659,7 +659,7 @@
       {
 	if (header && (file->flags & FLAG_VERB))
 	  {
-	    fprintf (stderr, "\n%*s USER        PID ACCESS COMMAND\n", NAME_FIELD, "");
+	    fprintf (stderr, _("\n%*s USER        PID ACCESS COMMAND\n"), NAME_FIELD, "");
 	    header = 0;
 	  }
 	length = 0;
--- psmisc-21.4/src/pstree.c.orig	2003-09-26 15:42:43.000000000 +0200
+++ psmisc-21.4/src/pstree.c	2003-12-11 22:30:22.626652232 +0100
@@ -726,9 +726,9 @@
   fprintf (stderr, _("    -s     show Flask SIDs\n"));
   fprintf (stderr, _("    -x     show Flask security contexts\n"));
 #endif /*FLASK_LINUX*/
-  fprintf (stderr, _("    -U     use UTF-8 (Unicode)) line drawing characters\n"));
+  fprintf (stderr, _("    -U     use UTF-8 (Unicode) line drawing characters\n"));
   fprintf (stderr, _("    -V     display version information\n"));
-  fprintf (stderr, _("    pid    start at pid, default 1 (init))\n"));
+  fprintf (stderr, _("    pid    start at pid, default 1 (init)\n"));
   fprintf (stderr, _("    user   show only trees rooted at processes of that user\n\n"));
   exit (1);
 }
--- psmisc-21.4/src/signals.c.orig	2003-04-03 06:29:21.000000000 +0200
+++ psmisc-21.4/src/signals.c	2003-12-11 22:31:37.258306496 +0100
@@ -8,6 +8,10 @@
 #include <string.h>
 #include <ctype.h>
 #include <signal.h>
+#include <libintl.h>
+#include <locale.h>
+#define _(String) gettext (String)
+
 #include "signals.h"
 
 
@@ -60,6 +64,6 @@
       break;
   if (walk->name)
     return walk->number;
-  fprintf (stderr, "%s: unknown signal; %s -l lists signals.\n", name, cmd);
+  fprintf (stderr, _("%s: unknown signal; %s -l lists signals.\n"), name, cmd);
   exit (1);
 }
--- psmisc-21.5/po/sv.po.orig	2004-04-30 05:39:16.000000000 +0200
+++ psmisc-21.5/po/sv.po	2004-07-16 00:46:45.057904656 +0200
@@ -285,8 +285,8 @@
 
 #: src/pstree.c:736
 #, c-format
-msgid "    -U     use UTF-8 (Unicode)) line drawing characters\n"
-msgstr "    -U     använd UTF-8 (Unicode)) linjeritningssymboler\n"
+msgid "    -U     use UTF-8 (Unicode) line drawing characters\n"
+msgstr "    -U     använd UTF-8 (Unicode) linjeritningssymboler\n"
 
 #: src/pstree.c:737
 #, c-format
@@ -295,8 +295,8 @@
 
 #: src/pstree.c:738
 #, c-format
-msgid "    pid    start at pid, default 1 (init))\n"
-msgstr "    pid    börja vid pid, standardvärde 1 (init))\n"
+msgid "    pid    start at pid, default 1 (init)\n"
+msgstr "    pid    börja vid pid, standardvärde 1 (init)\n"
 
 #: src/pstree.c:739
 #, c-format
