--- psmisc-21/src/Makefile.am	Wed May 15 13:57:17 2002
+++ psmisc-21/src/Makefile.am.org	Sun May 19 22:33:01 2002
@@ -1,15 +1,14 @@
 
-CFLAGS = -Wall -DLOCALEDIR=\"/usr/share/locale\"
-
-bin_PROGRAMS = fuser killall pstree
-
+DEFS = @DEFS@ -DLOCALEDIR=\"/usr/share/locale\"
+sbin_PROGRAMS = fuser
+bin_PROGRAMS = killall pstree
 fuser_SOURCES = fuser.c comm.h signals.c signals.h signames.h loop.h
 
 killall_SOURCES = killall.c comm.h signals.c signals.h signames.h
 
 pstree_SOURCES = pstree.c comm.h
 
-pstree_LDADD = @TERMCAP_LIB@
+pstree_LDADD = $(TERMCAP_LIB)
 
 BUILT_SOURCES = signames.h
 
--- psmisc-21.2/configure.in	Wed May 15 13:52:24 2002
+++ psmisc-21.2/configure.in.org	Sun May 19 22:34:29 2002
@@ -10,9 +10,10 @@
 AM_PROG_LIBTOOL
 
 dnl Checks for libraries.
-AC_CHECK_LIB(ncurses, tgetent, TERMCAP_LIB=-lncurses,
-  AC_CHECK_LIB(termcap, tgetent, TERMCAP_LIB=-ltermcap, 
-    AC_MSG_ERROR(Cannot find ncurses or termcap libraries) ) )
+AC_CHECK_LIB(tinfo, tgetent, TERMCAP_LIB=-ltinfo,
+  AC_CHECK_LIB(ncurses, tgetent, TERMCAP_LIB=-lncurses,
+    AC_CHECK_LIB(termcap, tgetent, TERMCAP_LIB=-ltermcap,
+      AC_MSG_ERROR(Cannot find ncurses or termcap libraries) ) ) )
 AC_SUBST(TERMCAP_LIB)
 
 dnl Checks for header files.
@@ -42,4 +42,4 @@
 dnl Checks for Large File System
 AC_SYS_LARGEFILE
 
-AC_OUTPUT(Makefile doc/Makefile src/Makefile po/Makefile.in po/Makefile.in )
+AC_OUTPUT(Makefile doc/Makefile src/Makefile po/Makefile.in )
