--- psmisc-21.3/doc/killall.1.selinux	2003-05-23 08:04:22.000000000 -0400
+++ psmisc-21.3/doc/killall.1	2003-08-28 14:01:12.000000000 -0400
@@ -4,7 +4,6 @@
 .SH SYNOPSIS
 .ad l
 .B killall
-.RB [ \-d , \-\-sid ]
 .RB [ \-c , \-\-context ] 
 .RB [ \-e , --exact ]
 .RB [ \-g , \-\-process-group ]
@@ -67,12 +66,9 @@
 any of the killed processes still exist and only returns if none are left.
 Note that \fBkillall\fP may wait forever if the signal was ignored, had no
 effect, or if the process stays in zombie state.
-.IP \fB\-d\fP
-(Flask only) Specify SID: kill only processes with given SID. Mutually exclusive
-with \fB-c\fP argument.  Must precede other arguments on command line.
-.IP \fB\-c\fP
-(Flask only) Specify security context: kill only processes with given security context.
-Mutually exclusive with \fB-d\fP.  Must precede other arguments on the command line.
+.IP \fB\-Z\fP
+(SELinux Only) Specify security context: kill only processes with given security context.
+Must precede other arguments on the command line.
 .SH FILES
 .nf
 /proc	location of the proc file system
--- psmisc-21.3/doc/pstree.1.selinux	2002-09-04 18:16:49.000000000 -0400
+++ psmisc-21.3/doc/pstree.1	2003-08-28 14:01:59.000000000 -0400
@@ -11,6 +11,7 @@
 .RB [ \-n ]
 .RB [ \-p ]
 .RB [ \-u ]
+.RB [ \-Z ]
 .RB [ \-G | \-U ]
 .RB [ \fIpid\fB | \fIuser\fB]
 .br
@@ -74,10 +75,8 @@
 with \fBecho -e '\\033%@'\fP
 .IP \fB\-V\fP
 Display version information.
-.IP \fB\-s\fP
-(Flask) Show Security ID (SID) for each process.
-.IP \fB\-x\fP
-(Flask) Show security context for each process.
+.IP \fB\-Z\fP
+(SELinux) Show security context for each process.
 .SH FILES
 .nf
 /proc	location of the proc file system
--- psmisc-21.3/src/killall.c.selinux	2003-05-23 08:47:47.000000000 -0400
+++ psmisc-21.3/src/killall.c	2003-08-28 14:03:54.000000000 -0400
@@ -21,10 +21,11 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <getopt.h>
-#ifdef FLASK_LINUX
-#include <selinux/fs_secure.h>
-#include <selinux/ss.h>
-#endif /*FLASK_LINUX*/
+
+#ifdef WITH_SELINUX
+#include <selinux/selinux.h>
+#endif /*WITH_SELINUX*/
+
 #include <libintl.h>
 #include <locale.h>
 #define _(String) gettext (String)
@@ -64,13 +65,13 @@
   return ch == 'y' || ch == 'Y';
 }
 
-#ifdef FLASK_LINUX
+#ifdef WITH_SELINUX
 static int
-kill_all(int signal, int names, char **namelist, security_id_t sid )
-#else  /*FLASK_LINUX*/
+kill_all(int signal, int names, char **namelist, security_context_t scontext )
+#else  /*WITH_SELINUX*/
 static int
 kill_all (int signal, int names, char **namelist)
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
 {
   DIR *dir;
   struct dirent *de;
@@ -85,11 +86,11 @@
   int empty, i, j, okay, length, got_long, error;
   int pids, max_pids, pids_killed;
   unsigned long found;
-#ifdef FLASK_LINUX
-  security_id_t lsid;
+#ifdef WITH_SELINUX
+  security_context_t lcontext=NULL;
 
   if ( names == 0 || ! namelist ) exit( 1 ); /* do the obvious thing...*/
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
 
   if (!(name_len = malloc (sizeof (int) * names)))
     {
@@ -102,19 +103,14 @@
 	sts[i].st_dev = 0;
 	name_len[i] = strlen (namelist[i]);
       }
-#ifdef FLASK_LINUX
-      else if (stat_secure(namelist[i],&sts[i], &lsid) < 0) {
-              perror(namelist[i]);
-              exit(1);
-          }
-#else  /*FLASK_LINUX*/
-    else if (stat (namelist[i], &sts[i]) < 0)
-      {
-	perror (namelist[i]);
-	exit (1);
-      }
-#endif /*FLASK_LINUX*/
-   } 
+    else {
+      if (stat (namelist[i], &sts[i]) < 0)
+	{
+	  perror (namelist[i]);
+	  exit (1);
+	}
+    }
+  } 
   self = getpid ();
   found = 0;
   if (!(dir = opendir (PROC_BASE)))
@@ -256,37 +252,39 @@
 	      else if (got_long ? strcmp (namelist[j], command) :
 		       strncmp (namelist[j], comm, COMM_LEN - 1))
 		continue;
-#ifdef FLASK_LINUX
-              if ( (int) sid > 0 ) {
-                if ( stat_secure(path, &st, &lsid) < 0 )
+#ifdef WITH_SELINUX
+              if ( scontext != NULL ) {
+                if ( getpidcon(pid_table[i], &lcontext) < 0 )
                   continue;
-                if ( lsid != sid )
+                if (strcmp(lcontext,scontext)!=0) {
+		  freecon(lcontext);
                   continue;
+		}
+		freecon(lcontext);
               }
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
 	    }
 	  else
 	    {
 	      if (asprintf (&path, PROC_BASE "/%d/exe", pid_table[i]) < 0)
 		continue;
-#ifdef FLASK_LINUX
-          if (stat_secure(path,&st,&lsid) < 0) {
-            free(path);
-            continue;
-          }
-          if (sts[j].st_dev != st.st_dev ||
-              sts[j].st_ino != st.st_ino ||
-              ((int) sid > 0 && (lsid != sid)) ) {
-            free(path);
-            continue;
-          }
-#else  /*FLASK_LINUX*/
+
 	      if (stat (path, &st) < 0) {
 		    free (path);
 		    continue;
 	      }
-#endif /*FLASK_LINUX*/
 	      free (path);
+#ifdef WITH_SELINUX
+              if ( scontext != NULL ) {
+                if ( getpidcon(pid_table[i], &lcontext) < 0 )
+                  continue;
+                if (strcmp(lcontext,scontext)!=0) {
+		  freecon(lcontext);
+                  continue;
+		}
+		freecon(lcontext);
+              }
+#endif /*WITH_SELINUX*/
 
 	      if (sts[j].st_dev != st.st_dev || sts[j].st_ino != st.st_ino)
 		continue;
@@ -383,13 +381,16 @@
 static void
 usage_killall (void)
 {
-#ifdef FLASK_LINUX
-  fprintf(stderr,"Usage: killall [-s sid] [-c context] [ -egiqvw ] [ -signal ] name ...\n");
-#else  /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+  fprintf(stderr,"Usage: killall [-Z context] [ -egiqvw ] [ -signal ] name ...\n");
+#else  /*WITH_SELINUX*/
   fprintf (stderr, "usage: killall [ OPTIONS ] [ -- ] name ...\n");
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
   fprintf (stderr, "       killall -l, --list\n");
   fprintf (stderr, "       killall -V --version\n\n");
+#ifdef WITH_SELINUX
+  fprintf (stderr, "  -Z,--context        kill only process(es) having scontext\n");
+#endif /*WITH_SELINUX*/
   fprintf (stderr, "  -e,--exact          require exact match for very long names\n");
   fprintf (stderr, "  -g,--process-group  kill process group instead of process\n");
   fprintf (stderr, "  -i,--interactive    ask for confirmation before killing\n");
@@ -399,11 +400,6 @@
   fprintf (stderr, "  -v,--verbose        report if the signal was successfully sent\n");
   fprintf (stderr, "  -V,--version        display version information\n");
   fprintf (stderr, "  -w,--wait           wait for processes to die\n\n");
-#ifdef FLASK_LINUX
-  fprintf (stderr, "  -d,--sid            kill only process(es) having sid\n");
-  fprintf (stderr, "  -c,--context        kill only process(es) having scontext\n");
-  fprintf(stderr, "   (-s, -c are mutually exclusive and must precede other arguments)\n\n");
-#endif /*FLASK_LINUX*/
 }
 
 
@@ -445,18 +441,17 @@
     {"signal", 1, NULL, 's'},
     {"verbose", 0, NULL, 'v'},
     {"wait", 0, NULL, 'w'},
-#ifdef FLASK_LINUX
-    {"Sid", 1, NULL, 'S'},
+#ifdef WITH_SELINUX
     {"context", 1, NULL, 'c'},
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
     {"version", 0, NULL, 'V'},
     {0,0,0,0 }};
 
-#ifdef FLASK_LINUX
-  security_id_t sid = -1;
+#ifdef WITH_SELINUX
+  security_context_t scontext = NULL;
 
   if ( argc < 2 ) usage(); /* do the obvious thing... */
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
 
   name = strrchr (*argv, '/');
   if (name)
@@ -472,8 +467,8 @@
   textdomain(PACKAGE);
 
   opterr = 0;
-#ifdef FLASK_LINUX
-  while ( (optc = getopt_long_only(argc,argv,"egilqs:vwd:c:V",options,NULL)) != EOF) {
+#ifdef WITH_SELINUX
+  while ( (optc = getopt_long_only(argc,argv,"egilqs:vwZ:V",options,NULL)) != EOF) {
 #else
   while ( (optc = getopt_long_only(argc,argv,"egilqs:vwV",options,NULL)) != EOF) {
 #endif
@@ -517,48 +512,14 @@
         print_version();
         return 0;
         break;
-#ifdef FLASK_LINUX
-      case 'd': {
-          char **buf, *calloc();
-          int strlen(), rv;
-          __u32 len;
-          security_id_t lsid;
-
-          buf = (char **) calloc(1, strlen(optarg));
-          if ( ! buf ) {
-             (void) fprintf(stderr, "%s: %s\n", name, strerror(errno));
-             return( 1 );
-          }
-
-	  lsid = strtol(optarg, buf, 0);
-          if ( **buf ) {
-              (void) fprintf(stderr, "%s: SID (%s) must be numeric\n", name, *argv);
-              (void) fflush(stderr);
-              return( 1 );
-          }
-
-          sid = (security_id_t) lsid;
-          /* sanity check */
-          len = strlen(optarg);
-          rv = security_sid_to_context(sid, buf, &len);
-          if ( rv < 0 && (errno != ENOSPC) ) {
-              (void) fprintf(stderr, "%s: security_sid_to_context(%d) %s\n", name, (int) sid, strerror(errno));
-              (void) fflush(stderr);
-              free(buf);
-              return( 1 );
-          }
-          free(buf);
-          break;
-      }
-      case 'c': {
-          if ( security_context_to_sid(optarg, strlen(optarg)+1, &sid) ) {
-              (void) fprintf(stderr, "%s: security_context_to_sid(%s): %s\n",
-                     name, optarg, strerror(errno));
-              (void) fflush(stderr);
-              return( 1 );
-          }
-      }
-#endif /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+	case 'Z': 
+	  if( is_selinux_enabled()) 
+	    scontext=optarg;
+	  else 
+	    fprintf(stderr, "Warning: -Z (--context) ignored. Requires an SELinux enabled kernel\n");
+	  break;
+#endif /*WITH_SELINUX*/
       case '?':
         /* Signal names are in uppercase, so check to see if the argv
          * is upper case */
@@ -590,9 +551,9 @@
     }
   argv = argv + myoptind;
   /*printf("sending signal %d to procs\n", sig_num);*/
-#ifdef FLASK_LINUX
-  return kill_all(sig_num,argc - myoptind, argv, sid);
-#else  /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+  return kill_all(sig_num,argc - myoptind, argv, scontext);
+#else  /*WITH_SELINUX*/
   return kill_all(sig_num,argc - myoptind, argv );
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
 }
--- psmisc-21.4/src/Makefile.am.orig	2003-11-28 12:22:16.000000000 +0100
+++ psmisc-21.4/src/Makefile.am	2003-12-11 00:27:12.834735104 +0100
@@ -9,11 +9,11 @@
 
 killall_SOURCES = killall.c comm.h signals.c signals.h signames.h
 
-killall_LDADD = @FLASK_LIB@
+killall_LDADD = @SELINUX_LIB@
 
 pstree_SOURCES = pstree.c comm.h
 
-pstree_LDADD = @TERMCAP_LIB@ @FLASK_LIB@
+pstree_LDADD = @TERMCAP_LIB@ @SELINUX_LIB@
 
 BUILT_SOURCES = signames.h
 
--- psmisc-21.4/src/pstree.c.orig	2003-09-26 15:42:43.000000000 +0200
+++ psmisc-21.4/src/pstree.c	2003-12-10 23:58:54.375940088 +0100
@@ -27,9 +27,9 @@
 
 #include "comm.h"
 
-#ifdef FLASK_LINUX
-#include <fs_secure.h>
-#endif /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+#include <selinux/selinux.h>
+#endif /*WITH_SELINUX*/
 
 #ifndef MAX_DEPTH
 #define MAX_DEPTH    100
@@ -58,9 +58,9 @@
   int argc;			/* with -a   : number of arguments, -1 if swapped    */
   pid_t pid;
   uid_t uid;
-#ifdef FLASK_LINUX
-  security_id_t sid;
-#endif /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+  security_context_t scontext;
+#endif /*WITH_SELINUX*/
   int highlight;
   struct _child *children;
   struct _proc *parent;
@@ -108,10 +108,9 @@
 static int width[MAX_DEPTH], more[MAX_DEPTH];
 static int print_args = 0, compact = 1, user_change = 0, pids = 0, by_pid = 0,
   trunc = 1, wait_end = 0;
-#ifdef FLASK_LINUX
-static int show_sids    = 0;
+#ifdef WITH_SELINUX
 static int show_scontext = 0;
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
 static int output_width = 132;
 static int cur_x = 1;
 static char last_char = 0;
@@ -161,38 +160,16 @@
   return digits;
 }
 
-#ifdef FLASK_LINUX
+#ifdef WITH_SELINUX
 static void 
-out_sid ( security_id_t sid )
+out_scontext ( security_context_t scontext )
 {
-  if ( (int) sid >= 0 )
-    out_int((int) sid);
-  else
-    out_string("??");
-}
-
-static void 
-out_scontext ( security_id_t sid )
-{
-  static char buf[256];
-  int security_sid_to_context();
-  int len = sizeof(buf);
-  int rv;
-
-  bzero(buf,256);
-
-  rv = security_sid_to_context((int)sid, buf, &len);
-  if ( rv ) {
-    out_string("`??\'"); /* punt */
-  }
-  else {
     out_string("`");
-    out_string(buf);
-    out_string("\'");
-  }
+    out_string(scontext);
+    out_string("'");
 }
-#endif /*FLASK_LINUX*/
-
+#endif /*WITH_SELINUX*/
+  
 
 static void
 out_newline (void)
@@ -216,13 +193,13 @@
   return walk;
 }
 
-#ifdef FLASK_LINUX
+#ifdef WITH_SELINUX
 static PROC *
-new_proc(const char *comm, pid_t pid, uid_t uid, security_id_t sid)
-#else  /*FLASK_LINUX*/
+new_proc(const char *comm, pid_t pid, uid_t uid, security_context_t scontext)
+#else  /*WITH_SELINUX*/
 static PROC *
 new_proc (const char *comm, pid_t pid, uid_t uid)
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
 {
   PROC *new;
 
@@ -235,9 +212,9 @@
   new->pid = pid;
   new->uid = uid;
   new->highlight = 0;
-#ifdef FLASK_LINUX
-  new->sid = sid;
-#endif /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+  new->scontext = scontext;
+#endif /*WITH_SELINUX*/
   new->children = NULL;
   new->parent = NULL;
   new->next = list;
@@ -306,24 +283,24 @@
     this->argv[i] = start = strchr (start, 0) + 1;
 }
 
-#ifdef FLASK_LINUX
+#ifdef WITH_SELINUX
 static void
 add_proc(const char *comm, pid_t pid, pid_t ppid, uid_t uid,
-         const char *args, int size, security_id_t sid)
-#else  /*FLASK_LINUX*/
+         const char *args, int size, security_context_t scontext)
+#else  /*WITH_SELINUX*/
 static void
 add_proc (const char *comm, pid_t pid, pid_t ppid, uid_t uid,
 	  const char *args, int size)
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
 {
   PROC *this, *parent;
 
   if (!(this = find_proc (pid)))
-#ifdef FLASK_LINUX
-    this = new_proc(comm, pid, uid, sid);
-#else  /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+    this = new_proc(comm, pid, uid, scontext);
+#else  /*WITH_SELINUX*/
     this = new_proc (comm, pid, uid);
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
   else
     {
       strcpy (this->comm, comm);
@@ -334,11 +311,11 @@
   if (pid == ppid)
     ppid = 0;
   if (!(parent = find_proc (ppid)))
-#ifdef FLASK_LINUX
-    parent = new_proc("?", ppid, 0, sid);
-#else  /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+    parent = new_proc("?", ppid, 0, scontext);
+#else  /*WITH_SELINUX*/
     parent = new_proc ("?", ppid, 0);
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
   add_child (parent, this);
   this->parent = parent;
 }
@@ -430,25 +407,17 @@
       else
 	(void) out_int (current->uid);
     }
-#ifdef FLASK_LINUX
-  if ( show_sids ) {
-    out_char (info++ ? ',' : '(');
-    out_sid(current->sid);
-  }
+#ifdef WITH_SELINUX
   if ( show_scontext ) {
     out_char (info++ ? ',' : '(');
-    out_scontext(current->sid);
+    out_scontext(current->scontext);
   }
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
   if ((swapped && print_args && current->argc < 0) || (!swapped && info))
     out_char (')');
   if (current->highlight && (tmp = tgetstr ("me", NULL)))
     tputs (tmp, 1, putchar);
-#ifdef FLASK_LINUX
-  if (show_scontext || print_args)
-#else  /*FLASK_LINUX*/
   if (print_args)
-#endif /*FLASK_LINUX*/
     {
       for (i = 0; i < current->argc; i++)
 	{
@@ -473,20 +442,20 @@
 	    }
 	}
     }
-#ifdef FLASK_LINUX
+#ifdef WITH_SELINUX
   if ( show_scontext || print_args || ! current->children )
-#else  /*FLASK_LINUX*/
+#else  /*WITH_SELINUX*/
   if (print_args || !current->children)
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
     {
       while (closing--)
 	out_char (']');
       out_newline ();
-#ifdef FLASK_LINUX
+#ifdef WITH_SELINUX
       if ( show_scontext || print_args )
-#else /*FLASK_LINUX*/
+#else /*WITH_SELINUX*/
       if (print_args)
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
 	{
 	  more[level] = !last;
 	  width[level] = swapped + (comm_len > 1 ? 0 : -1);
@@ -576,9 +545,10 @@
   pid_t pid, ppid;
   int fd, size;
   int empty;
-#ifdef FLASK_LINUX
-  security_id_t sid = -1;
-#endif /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+  security_context_t scontext = NULL;
+  int selinux_enabled=is_selinux_enabled();
+#endif /*WITH_SELINUX*/
 
   if (!print_args)
     buffer = NULL;
@@ -603,11 +573,15 @@
 	  {
 	    empty = 0;
 	    sprintf (path, "%s/%d", PROC_BASE, pid);
-#ifdef FLASK_LINUX
-            if (fstat_secure(fileno(file),&st,&sid) < 0)
-#else /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+	    if (selinux_enabled)
+	      if (getpidcon(pid,&scontext) < 0)
+		{
+		  perror (path);
+		  exit (1);
+		}
+#endif /*WITH_SELINUX*/
             if (stat (path, &st) < 0)
-#endif /*FLASK_LINUX*/
 	    {
 		perror (path);
 		exit (1);
@@ -632,11 +606,11 @@
 		 &ppid) == 4)
  */
 		if (!print_args)
-#ifdef FLASK_LINUX
-		  add_proc(comm, pid, ppid, st.st_uid, NULL, 0, sid);
-#else  /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+		  add_proc(comm, pid, ppid, st.st_uid, NULL, 0, scontext);
+#else  /*WITH_SELINUX*/
 		  add_proc (comm, pid, ppid, st.st_uid, NULL, 0);
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
 		else
 		  {
 		    sprintf (path, "%s/%d/cmdline", PROC_BASE, pid);
@@ -653,11 +627,11 @@
 		    (void) close (fd);
 		    if (size)
 		      buffer[size++] = 0;
-#ifdef FLASK_LINUX
-		    add_proc(comm, pid, ppid, st.st_uid, buffer, size, sid);
-#else  /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+		    add_proc(comm, pid, ppid, st.st_uid, buffer, size, scontext);
+#else  /*WITH_SELINUX*/
 		    add_proc (comm, pid, ppid, st.st_uid, buffer, size);
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
 		  }
 		}
 	      }
@@ -696,11 +670,11 @@
 	cmd = comm;
       if (*cmd == '-')
 	cmd++;
-#ifdef FLASK_LINUX
-      add_proc(cmd, pid, ppid, uid, NULL, 0, -1);
-#else  /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+      add_proc(cmd, pid, ppid, uid, NULL, 0, NULL);
+#else  /*WITH_SELINUX*/
       add_proc (cmd, pid, ppid, uid, NULL, 0);
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
     }
 }
 
@@ -722,10 +696,9 @@
   fprintf (stderr, _("    -n     sort output by PID\n"));
   fprintf (stderr, _("    -p     show PIDs; implies -c\n"));
   fprintf (stderr, _("    -u     show uid transitions\n"));
-#ifdef FLASK_LINUX
-  fprintf (stderr, _("    -s     show Flask SIDs\n"));
-  fprintf (stderr, _("    -x     show Flask security contexts\n"));
-#endif /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+  fprintf (stderr, _("    -Z     show SELinux security contexts\n"));
+#endif /*WITH_SELINUX*/
   fprintf (stderr, _("    -U     use UTF-8 (Unicode)) line drawing characters\n"));
   fprintf (stderr, _("    -V     display version information\n"));
   fprintf (stderr, _("    pid    start at pid, default 1 (init))\n"));
@@ -771,11 +744,11 @@
   }
 
 
-#ifdef FLASK_LINUX
-  while ((c = getopt (argc, argv, "acGhH:npluUVsx")) != EOF)
-#else  /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+  while ((c = getopt (argc, argv, "acGhH:npluUVZ")) != EOF)
+#else  /*WITH_SELINUX*/
   while ((c = getopt (argc, argv, "acGhH:npluUV")) != EOF)
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
     switch (c)
       {
       case 'a':
@@ -832,14 +805,14 @@
       case 'V':
       print_version();
 	return 0;
-#ifdef FLASK_LINUX
-      case 's':
-        show_sids = 1;
-        break;
-      case 'x':
-        show_scontext = 1;
+#ifdef WITH_SELINUX
+      case 'Z':
+	if (is_selinux_enabled())
+	  show_scontext = 1;
+	else
+	  fprintf(stderr, "Warning: -Z ignored. Requires anx SELinux enabled kernel\n");
         break;
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
       default:
 	usage ();
       }
--- psmisc-21.3/configure.in.selinux	2003-05-28 19:45:24.000000000 -0400
+++ psmisc-21.3/configure.in	2003-08-28 14:00:20.000000000 -0400
@@ -9,14 +9,14 @@
 AM_PROG_LIBTOOL
 
 dnl checks for options
-AC_SUBST(FLASK_LINUX)
-AC_ARG_ENABLE(flask,[  --enable-flask  Enable Security-Enhanced Linux features],
-	AC_DEFINE([FLASK_LINUX],1,[Use Security-Enhanced Linux features])
-	AC_CHECK_LIB(secure, avc_toggle, FLASK_LIB=-lsecure,
-	   AC_MSG_ERROR(Cannot find selinux/ secure static library)
+AC_SUBST(WITH_SELINUX)
+AC_ARG_ENABLE(selinux,[  --enable-selinux  Enable Security-Enhanced Linux features],
+	AC_DEFINE([WITH_SELINUX],1,[Use Security-Enhanced Linux features])
+	AC_CHECK_LIB(selinux,getfilecon, SELINUX_LIB=-lselinux,
+	   AC_MSG_ERROR(Cannot find selinux secure static library)
 	)
 ,)
-AC_SUBST(FLASK_LIB)
+AC_SUBST(SELINUX_LIB)
 
 dnl Checks for libraries.
 AC_CHECK_LIB(ncurses, tgetent, TERMCAP_LIB=-lncurses,
--- psmisc-21.4/po/pl.po.orig	2003-12-11 00:50:15.000000000 +0100
+++ psmisc-21.4/po/pl.po	2003-12-11 00:52:29.458173296 +0100
@@ -6,41 +6,41 @@
 msgid ""
 msgstr ""
 "Project-Id-Version: psmisc 21.2\n"
-"POT-Creation-Date: 2003-12-11 00:33+0100\n"
+"Report-Msgid-Bugs-To: csmall@small.dropbear.id.au\n"
+"POT-Creation-Date: 2003-12-11 00:52+0100\n"
 "PO-Revision-Date: 2003-12-11 00:35+0100\n"
 "Last-Translator: Jakub Bogusz <qboosh@pld-linux.org>\n"
 "Language-Team: Polish <translation-team-pl@lists.sourceforge.net>\n"
 "MIME-Version: 1.0\n"
 "Content-Type: text/plain; charset=ISO-8859-2\n"
 "Content-Transfer-Encoding: 8bit\n"
-"Report-Msgid-Bugs-To: csmall@small.dropbear.id.au\n"
 
-#: src/killall.c:51
+#: src/killall.c:52
 #, c-format
 msgid "Kill %s(%s%d) ? (y/n) "
 msgstr "Zabiæ %s(%s%d)? (y/n) "
 
-#: src/killall.c:239
+#: src/killall.c:235
 #, c-format
 msgid "skipping partial match %s(%d)\n"
 msgstr "pominiêto czê¶ciowe dopasowanie %s(%d)\n"
 
-#: src/killall.c:325
+#: src/killall.c:323
 #, c-format
 msgid "Killed %s(%s%d) with signal %d\n"
 msgstr "Zabito %s(%s%d) sygna³em %d\n"
 
-#: src/killall.c:337 src/fuser.c:503 src/pstree.c:674
+#: src/killall.c:335 src/fuser.c:503 src/pstree.c:648
 #, c-format
 msgid "%s is empty (not mounted ?)\n"
 msgstr "%s jest pusty (nie podmontowany?)\n"
 
-#: src/killall.c:343
+#: src/killall.c:341
 #, c-format
 msgid "%s: no process killed\n"
 msgstr "%s: nie zabito ¿adnego procesu\n"
 
-#: src/killall.c:588
+#: src/killall.c:549
 #, c-format
 msgid "Maximum number of names is %d\n"
 msgstr "Maksymalna liczba nazw to %d\n"
@@ -181,22 +181,22 @@
 msgid "%s/%s: invalid specificiation\n"
 msgstr "%s/%s: b³êdne okre¶lenie\n"
 
-#: src/pstree.c:378
+#: src/pstree.c:355
 #, c-format
 msgid "MAX_DEPTH not big enough.\n"
 msgstr "MAX_DEPTH jest zbyt ma³e.\n"
 
-#: src/pstree.c:713
+#: src/pstree.c:687
 #, c-format
 msgid "usage: pstree [ -a ] [ -c ] [ -h | -H pid ] [ -l ] [ -n ] [ -p ] [ -u ]\n"
 msgstr "sk³adnia: pstree [ -a ] [ -c ] [ -h | -H pid ] [ -l ] [ -n ] [ -p ] [ -u ]\n"
 
-#: src/pstree.c:714
+#: src/pstree.c:688
 #, c-format
 msgid "              [ -G | -U ] [ pid | user]\n"
 msgstr "              [ -G | -U ] [ pid | u¿ytkownik ]\n"
 
-#: src/pstree.c:715
+#: src/pstree.c:689
 #, c-format
 msgid ""
 "       pstree -V\n"
@@ -205,77 +205,72 @@
 "       pstree -V\n"
 "\n"
 
-#: src/pstree.c:716
+#: src/pstree.c:690
 #, c-format
 msgid "    -a     show command line arguments\n"
 msgstr "    -a     wy¶wietlenie argumentów linii poleceñ\n"
 
-#: src/pstree.c:717
+#: src/pstree.c:691
 #, c-format
 msgid "    -c     don't compact identical subtrees\n"
 msgstr "    -c     nie scalanie identycznych poddrzew\n"
 
-#: src/pstree.c:718
+#: src/pstree.c:692
 #, c-format
 msgid "    -h     highlight current process and its ancestors\n"
 msgstr "    -h     pod¶wietlenie aktualnego procesu i jego przodków\n"
 
-#: src/pstree.c:719
+#: src/pstree.c:693
 #, c-format
 msgid "    -H pid highlight process \"pid\" and its ancestors\n"
 msgstr "    -H pid pod¶wietlenie procesu \"pid\" i jego przodków\n"
 
-#: src/pstree.c:720
+#: src/pstree.c:694
 #, c-format
 msgid "    -G     use VT100 line drawing characters\n"
 msgstr "    -G     u¿ycie znaków ramek VT100\n"
 
-#: src/pstree.c:721
+#: src/pstree.c:695
 #, c-format
 msgid "    -l     don't truncate long lines\n"
 msgstr "    -l     nie obcinanie d³ugich linii\n"
 
-#: src/pstree.c:722
+#: src/pstree.c:696
 #, c-format
 msgid "    -n     sort output by PID\n"
 msgstr "    -n     sortowanie wyj¶cia wg PID-u\n"
 
-#: src/pstree.c:723
+#: src/pstree.c:697
 #, c-format
 msgid "    -p     show PIDs; implies -c\n"
 msgstr "    -p     wy¶wietlanie PID-ów; w³±cza tak¿e -c\n"
 
-#: src/pstree.c:724
+#: src/pstree.c:698
 #, c-format
 msgid "    -u     show uid transitions\n"
 msgstr "    -u     wy¶wietlanie zmian uidów\n"
 
-#: src/pstree.c:726
+#: src/pstree.c:700
 #, c-format
-msgid "    -s     show Flask SIDs\n"
-msgstr "    -s     wy¶wietlanie SID-ów Flask\n"
+msgid "    -Z     show SELinux security contexts\n"
+msgstr "    -Z     wy¶wietlanie kontekstów bezpieczeñstwa SELinux\n"
 
-#: src/pstree.c:727
-#, c-format
-msgid "    -x     show Flask security contexts\n"
-msgstr "    -x     wy¶wietlanie kontekstów bezpieczeñstwa Flask\n"
-
-#: src/pstree.c:729
+#: src/pstree.c:702
 #, c-format
 msgid "    -U     use UTF-8 (Unicode)) line drawing characters\n"
 msgstr "    -U     u¿ycie znaków ramek UTF-8 (Unicode)\n"
 
-#: src/pstree.c:730
+#: src/pstree.c:703
 #, c-format
 msgid "    -V     display version information\n"
 msgstr "    -V     wy¶wietlenie informacji o wersji\n"
 
-#: src/pstree.c:731
+#: src/pstree.c:704
 #, c-format
 msgid "    pid    start at pid, default 1 (init))\n"
 msgstr "    pid    rozpoczêcie od tego pida, domy¶lnie 1 (init)\n"
 
-#: src/pstree.c:732
+#: src/pstree.c:705
 #, c-format
 msgid ""
 "    user   show only trees rooted at processes of that user\n"
@@ -284,12 +279,12 @@
 " u¿ytkownik  tylko drzewa zaczynaj±ce siê od procesów tego u¿ytkownika\n"
 "\n"
 
-#: src/pstree.c:738
+#: src/pstree.c:711
 #, c-format
 msgid "pstree (psmisc) %s\n"
 msgstr "pstree (psmisc) %s\n"
 
-#: src/pstree.c:739
+#: src/pstree.c:712
 #, c-format
 msgid ""
 "Copyright (C) 1993-2002 Werner Almesberger and Craig Small\n"
@@ -298,47 +293,47 @@
 "Copyright (C) 1993-2002 Werner Almesberger i Craig Small\n"
 "\n"
 
-#: src/pstree.c:740
+#: src/pstree.c:713
 #, c-format
 msgid "pstree comes with ABSOLUTELY NO WARRANTY.\n"
 msgstr "pstree jest rozpowszechniane BEZ ¯ADNEJ GWARANCJI.\n"
 
-#: src/pstree.c:741
+#: src/pstree.c:714
 #, c-format
 msgid "This is free software, and you are welcome to redistribute it under the terms\n"
 msgstr "To oprogramowanie jest darmowe i mo¿e byæ dystrybuowane na warunkach\n"
 
-#: src/pstree.c:742
+#: src/pstree.c:715
 #, c-format
 msgid "of the GNU General Public License.\n"
 msgstr "Powszechnej Licencji Publicznej GNU (General Public License).\n"
 
-#: src/pstree.c:743
+#: src/pstree.c:716
 #, c-format
 msgid "For more information about these matters, see the files named COPYING.\n"
 msgstr "Wiêcej informacji znajduje siê w pliku o nazwie COPYING.\n"
 
-#: src/pstree.c:803
+#: src/pstree.c:776
 #, c-format
 msgid "TERM is not set\n"
 msgstr "TERM nie ustawiony\n"
 
-#: src/pstree.c:808
+#: src/pstree.c:781
 #, c-format
 msgid "Can't get terminal capabilities\n"
 msgstr "Nie mo¿na odczytaæ mo¿liwo¶ci terminala\n"
 
-#: src/pstree.c:854
+#: src/pstree.c:827
 #, c-format
 msgid "No such user name: %s\n"
 msgstr "Nie ma u¿ytkownika o takiej nazwie: %s\n"
 
-#: src/pstree.c:878
+#: src/pstree.c:851
 #, c-format
 msgid "No processes found.\n"
 msgstr "Nie znaleziono procesów.\n"
 
-#: src/pstree.c:883
+#: src/pstree.c:856
 #, c-format
 msgid "Press return to close\n"
 msgstr "Naci¶niêcie return zamknie program\n"
