--- psmisc-21.6/doc/killall.1.orig	2004-12-15 12:15:52.000000000 +0100
+++ psmisc-21.6/doc/killall.1	2005-03-20 22:22:36.420168640 +0100
@@ -4,7 +4,6 @@
 .SH SYNOPSIS
 .ad l
 .B killall
-.RB [ \-d , \-\-sid ]
 .RB [ \-c , \-\-context ]
 .RB [ \-e , \-\-exact ]
 .RB [ \-g , \-\-process\-group ]
@@ -70,13 +69,9 @@
 any of the killed processes still exist and only returns if none are left.
 Note that \fBkillall\fP may wait forever if the signal was ignored, had no
 effect, or if the process stays in zombie state.
-.IP \fB\-d\fP
-(Flask only) Specify SID: kill only processes with given SID. Mutually exclusive
-with \fB\-c\fP argument. Must precede other arguments on command line.
-.IP \fB\-c\fP
-(Flask only) Specify security context: kill only processes with given security
-context. Mutually exclusive with \fB\-d\fP. Must precede other arguments on
-the command line.
+.IP \fB\-Z\fP
+(SELinux Only) Specify security context: kill only processes with given security context.
+Must precede other arguments on the command line.
 .SH FILES
 .nf
 /proc	location of the proc file system
--- psmisc-21.5/doc/pstree.1.orig	2004-04-24 14:44:27.000000000 +0200
+++ psmisc-21.5/doc/pstree.1	2004-07-16 07:50:24.316589960 +0200
@@ -12,6 +12,7 @@
 .RB [ \-p ]
 .RB [ \-u ]
 .RB [ \-A | \-G | \-U ]
+.RB [ \-Z ]
 .RB [ \fIpid\fB | \fIuser\fB]
 .br
 .B pstree
@@ -81,10 +82,8 @@
 with \fBecho -e '\\033%@'\fP
 .IP \fB\-V\fP
 Display version information.
-.IP \fB\-s\fP
-(Flask) Show Security ID (SID) for each process.
-.IP \fB\-x\fP
-(Flask) Show security context for each process.
+.IP \fB\-Z\fP
+(SELinux) Show security context for each process.
 .SH FILES
 .nf
 /proc	location of the proc file system
--- psmisc-21.6/src/killall.c.orig	2004-12-10 12:31:29.000000000 +0100
+++ psmisc-21.6/src/killall.c	2005-03-20 22:28:07.514834600 +0100
@@ -21,10 +21,11 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <getopt.h>
-#ifdef FLASK_LINUX
-#include <selinux/fs_secure.h>
-#include <selinux/ss.h>
-#endif /*FLASK_LINUX*/
+
+
+#ifdef WITH_SELINUX
+#include <selinux/selinux.h>
+#endif /*WITH_SELINUX*/
 
 #include "i18n.h"
 #include "comm.h"
@@ -70,13 +71,13 @@
   /* Never should get here */
 }
 
-#ifdef FLASK_LINUX
+#ifdef WITH_SELINUX
 static int
-kill_all(int signal, int names, char **namelist, security_id_t sid )
-#else  /*FLASK_LINUX*/
+kill_all(int signal, int names, char **namelist, security_context_t scontext )
+#else  /*WITH_SELINUX*/
 static int
 kill_all (int signal, int names, char **namelist)
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
 {
   DIR *dir;
   struct dirent *de;
@@ -91,11 +92,11 @@
   int empty, i, j, okay, length, got_long, error;
   int pids, max_pids, pids_killed;
   unsigned long found;
-#ifdef FLASK_LINUX
-  security_id_t lsid;
+#ifdef WITH_SELINUX
+  security_context_t lcontext=NULL;
 
   if ( names == 0 || ! namelist ) exit( 1 ); /* do the obvious thing...*/
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
 
   if (!(name_len = malloc (sizeof (int) * names)))
     {
@@ -108,19 +109,14 @@
 	sts[i].st_dev = 0;
 	name_len[i] = strlen (namelist[i]);
       }
-#ifdef FLASK_LINUX
-      else if (stat_secure(namelist[i],&sts[i], &lsid) < 0) {
-              perror(namelist[i]);
-              exit(1);
-          }
-#else  /*FLASK_LINUX*/
-    else if (stat (namelist[i], &sts[i]) < 0)
-      {
-	perror (namelist[i]);
-	exit (1);
-      }
-#endif /*FLASK_LINUX*/
-   } 
+    else {
+      if (stat (namelist[i], &sts[i]) < 0)
+	{
+	  perror (namelist[i]);
+	  exit (1);
+	}
+    }
+  } 
   self = getpid ();
   found = 0;
   if (!(dir = opendir (PROC_BASE)))
@@ -270,37 +266,39 @@
 	      else if (got_long ? strcmp (namelist[j], command) :
 		       strncmp (namelist[j], comm, COMM_LEN - 1))
 		continue;
-#ifdef FLASK_LINUX
-              if ( (int) sid > 0 ) {
-                if ( stat_secure(path, &st, &lsid) < 0 )
+#ifdef WITH_SELINUX
+              if ( scontext != NULL ) {
+                if ( getpidcon(pid_table[i], &lcontext) < 0 )
                   continue;
-                if ( lsid != sid )
+                if (strcmp(lcontext,scontext)!=0) {
+		  freecon(lcontext);
                   continue;
+		}
+		freecon(lcontext);
               }
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
 	    }
 	  else
 	    {
 	      if (asprintf (&path, PROC_BASE "/%d/exe", pid_table[i]) < 0)
 		continue;
-#ifdef FLASK_LINUX
-          if (stat_secure(path,&st,&lsid) < 0) {
-            free(path);
-            continue;
-          }
-          if (sts[j].st_dev != st.st_dev ||
-              sts[j].st_ino != st.st_ino ||
-              ((int) sid > 0 && (lsid != sid)) ) {
-            free(path);
-            continue;
-          }
-#else  /*FLASK_LINUX*/
+
 	      if (stat (path, &st) < 0) {
 		    free (path);
 		    continue;
 	      }
-#endif /*FLASK_LINUX*/
 	      free (path);
+#ifdef WITH_SELINUX
+              if ( scontext != NULL ) {
+                if ( getpidcon(pid_table[i], &lcontext) < 0 )
+                  continue;
+                if (strcmp(lcontext,scontext)!=0) {
+		  freecon(lcontext);
+                  continue;
+		}
+		freecon(lcontext);
+              }
+#endif /*WITH_SELINUX*/
 
 	      if (sts[j].st_dev != st.st_dev || sts[j].st_ino != st.st_ino)
 		continue;
@@ -397,13 +395,13 @@
 static void
 usage_killall (void)
 {
-#ifdef FLASK_LINUX
+#ifdef WITH_SELINUX
   fprintf(stderr, _(
-    "usage: killall [-s sid] [-c context] [ -egiqvw ] [ -signal ] name ...\n"));
-#else  /*FLASK_LINUX*/
+    "usage: killall [-Z context] [ -egiqvw ] [ -signal ] name ...\n"));
+#else  /*WITH_SELINUX*/
   fprintf(stderr, _(
     "usage: killall [ OPTIONS ] [ -- ] name ...\n"));
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
   fprintf(stderr, _(
     "       killall -l, --list\n"
     "       killall -V --version\n\n"
@@ -417,13 +415,12 @@
     "  -v,--verbose        report if the signal was successfully sent\n"
     "  -V,--version        display version information\n"
     "  -w,--wait           wait for processes to die\n\n"));
-#ifdef FLASK_LINUX
+#ifdef WITH_SELINUX
   fprintf(stderr, _(
-    "  -d,--sid            kill only process(es) having sid\n"
-    "  -c,--context        kill only process(es) having scontext\n"
-    "   (-s, -c are mutually exclusive and must precede other arguments)\n\n"
+    "  -Z,--context        kill only process(es) having scontext\n"
+    "   (-Z must precede other arguments)\n\n"
     ));
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
 }
 
 
@@ -468,18 +465,17 @@
     {"signal", 1, NULL, 's'},
     {"verbose", 0, NULL, 'v'},
     {"wait", 0, NULL, 'w'},
-#ifdef FLASK_LINUX
-    {"Sid", 1, NULL, 'd'},
+#ifdef WITH_SELINUX
     {"context", 1, NULL, 'c'},
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
     {"version", 0, NULL, 'V'},
     {0,0,0,0 }};
 
-#ifdef FLASK_LINUX
-  security_id_t sid = -1;
+#ifdef WITH_SELINUX
+  security_context_t scontext = NULL;
 
   if ( argc < 2 ) usage(); /* do the obvious thing... */
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
 
   name = strrchr (*argv, '/');
   if (name)
@@ -497,8 +493,8 @@
 #endif
 
   opterr = 0;
-#ifdef FLASK_LINUX
-  while ( (optc = getopt_long_only(argc,argv,"egilqs:vwd:c:VI",options,NULL)) != EOF) {
+#ifdef WITH_SELINUX
+  while ( (optc = getopt_long_only(argc,argv,"egilqs:vwZ:VI",options,NULL)) != EOF) {
 #else
   while ( (optc = getopt_long_only(argc,argv,"egilqs:vwVI",options,NULL)) != EOF) {
 #endif
@@ -545,50 +541,14 @@
         print_version();
         return 0;
         break;
-#ifdef FLASK_LINUX
-      case 'd': {
-          char **buf, *calloc();
-          int strlen(), rv;
-          __u32 len;
-          security_id_t lsid;
-
-          buf = (char **) calloc(1, strlen(optarg));
-          if ( ! buf ) {
-             (void) fprintf(stderr, "%s: %s\n", name, strerror(errno));
-             return( 1 );
-          }
-
-	  lsid = strtol(optarg, buf, 0);
-          if ( **buf ) {
-              (void) fprintf(stderr, _("%s: SID (%s) must be numeric\n"),
-			     name, *argv);
-              (void) fflush(stderr);
-              return( 1 );
-          }
-
-          sid = (security_id_t) lsid;
-          /* sanity check */
-          len = strlen(optarg);
-          rv = security_sid_to_context(sid, buf, &len);
-          if ( rv < 0 && (errno != ENOSPC) ) {
-              (void) fprintf(stderr, "%s: security_sid_to_context(%d) %s\n",
-			     name, (int) sid, strerror(errno));
-              (void) fflush(stderr);
-              free(buf);
-              return( 1 );
-          }
-          free(buf);
-          break;
-      }
-      case 'c': {
-          if ( security_context_to_sid(optarg, strlen(optarg)+1, &sid) ) {
-              (void) fprintf(stderr, "%s: security_context_to_sid(%s): %s\n",
-                     name, optarg, strerror(errno));
-              (void) fflush(stderr);
-              return( 1 );
-          }
-      }
-#endif /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+	case 'Z': 
+	  if( is_selinux_enabled() > 0)
+	    scontext=optarg;
+	  else 
+	    fprintf(stderr, _("Warning: -Z (--context) ignored. Requires an SELinux enabled kernel\n"));
+	  break;
+#endif /*WITH_SELINUX*/
       case '?':
         /* Signal names are in uppercase, so check to see if the argv
          * is upper case */
@@ -616,9 +576,9 @@
     }
   argv = argv + myoptind;
   /*printf("sending signal %d to procs\n", sig_num);*/
-#ifdef FLASK_LINUX
-  return kill_all(sig_num,argc - myoptind, argv, sid);
-#else  /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+  return kill_all(sig_num,argc - myoptind, argv, scontext);
+#else  /*WITH_SELINUX*/
   return kill_all(sig_num,argc - myoptind, argv );
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
 }
--- psmisc-21.6/src/Makefile.am.orig	2005-02-23 03:25:00.000000000 +0100
+++ psmisc-21.6/src/Makefile.am	2005-03-20 22:29:19.228932400 +0100
@@ -7,11 +7,11 @@
 
 killall_SOURCES = killall.c comm.h signals.c signals.h i18n.h
 
-killall_LDADD = @FLASK_LIB@
+killall_LDADD = @SELINUX_LIB@
 
 pstree_SOURCES = pstree.c comm.h i18n.h
 
-pstree_LDADD = @TERMCAP_LIB@ @FLASK_LIB@
+pstree_LDADD = @TERMCAP_LIB@ @SELINUX_LIB@
 
 BUILT_SOURCES = signames.h
 
--- psmisc-21.6/src/pstree.c.orig	2004-12-15 12:15:52.000000000 +0100
+++ psmisc-21.6/src/pstree.c	2005-03-20 22:31:36.737027992 +0100
@@ -26,9 +26,9 @@
 #include "i18n.h"
 #include "comm.h"
 
-#ifdef FLASK_LINUX
-#include <fs_secure.h>
-#endif /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+#include <selinux/selinux.h>
+#endif /*WITH_SELINUX*/
 
 #ifndef MAX_DEPTH
 #define MAX_DEPTH    100
@@ -57,9 +57,9 @@
   int argc;			/* with -a   : number of arguments, -1 if swapped    */
   pid_t pid;
   uid_t uid;
-#ifdef FLASK_LINUX
-  security_id_t sid;
-#endif /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+  security_context_t scontext;
+#endif /*WITH_SELINUX*/
   int highlight;
   struct _child *children;
   struct _proc *parent;
@@ -107,10 +107,9 @@
 static int width[MAX_DEPTH], more[MAX_DEPTH];
 static int print_args = 0, compact = 1, user_change = 0, pids = 0, by_pid = 0,
   trunc = 1, wait_end = 0;
-#ifdef FLASK_LINUX
-static int show_sids    = 0;
+#ifdef WITH_SELINUX
 static int show_scontext = 0;
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
 static int output_width = 132;
 static int cur_x = 1;
 static char last_char = 0;
@@ -160,38 +159,16 @@
   return digits;
 }
 
-#ifdef FLASK_LINUX
-static void 
-out_sid ( security_id_t sid )
-{
-  if ( (int) sid >= 0 )
-    out_int((int) sid);
-  else
-    out_string("??");
-}
-
+#ifdef WITH_SELINUX
 static void 
-out_scontext ( security_id_t sid )
+out_scontext ( security_context_t scontext )
 {
-  static char buf[256];
-  int security_sid_to_context();
-  int len = sizeof(buf);
-  int rv;
-
-  bzero(buf,256);
-
-  rv = security_sid_to_context((int)sid, buf, &len);
-  if ( rv ) {
-    out_string("`??\'"); /* punt */
-  }
-  else {
     out_string("`");
-    out_string(buf);
-    out_string("\'");
-  }
+    out_string(scontext);
+    out_string("'");
 }
-#endif /*FLASK_LINUX*/
-
+#endif /*WITH_SELINUX*/
+  
 
 static void
 out_newline (void)
@@ -215,13 +192,13 @@
   return walk;
 }
 
-#ifdef FLASK_LINUX
+#ifdef WITH_SELINUX
 static PROC *
-new_proc(const char *comm, pid_t pid, uid_t uid, security_id_t sid)
-#else  /*FLASK_LINUX*/
+new_proc(const char *comm, pid_t pid, uid_t uid, security_context_t scontext)
+#else  /*WITH_SELINUX*/
 static PROC *
 new_proc (const char *comm, pid_t pid, uid_t uid)
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
 {
   PROC *new;
 
@@ -234,9 +211,9 @@
   new->pid = pid;
   new->uid = uid;
   new->highlight = 0;
-#ifdef FLASK_LINUX
-  new->sid = sid;
-#endif /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+  new->scontext = scontext;
+#endif /*WITH_SELINUX*/
   new->children = NULL;
   new->parent = NULL;
   new->next = list;
@@ -305,24 +282,24 @@
     this->argv[i] = start = strchr (start, 0) + 1;
 }
 
-#ifdef FLASK_LINUX
+#ifdef WITH_SELINUX
 static void
 add_proc(const char *comm, pid_t pid, pid_t ppid, uid_t uid,
-         const char *args, int size, security_id_t sid)
-#else  /*FLASK_LINUX*/
+         const char *args, int size, security_context_t scontext)
+#else  /*WITH_SELINUX*/
 static void
 add_proc (const char *comm, pid_t pid, pid_t ppid, uid_t uid,
 	  const char *args, int size)
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
 {
   PROC *this, *parent;
 
   if (!(this = find_proc (pid)))
-#ifdef FLASK_LINUX
-    this = new_proc(comm, pid, uid, sid);
-#else  /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+    this = new_proc(comm, pid, uid, scontext);
+#else  /*WITH_SELINUX*/
     this = new_proc (comm, pid, uid);
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
   else
     {
       strcpy (this->comm, comm);
@@ -333,11 +310,11 @@
   if (pid == ppid)
     ppid = 0;
   if (!(parent = find_proc (ppid)))
-#ifdef FLASK_LINUX
-    parent = new_proc("?", ppid, 0, sid);
-#else  /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+    parent = new_proc("?", ppid, 0, scontext);
+#else  /*WITH_SELINUX*/
     parent = new_proc ("?", ppid, 0);
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
   add_child (parent, this);
   this->parent = parent;
 }
@@ -429,25 +406,17 @@
       else
 	(void) out_int (current->uid);
     }
-#ifdef FLASK_LINUX
-  if ( show_sids ) {
-    out_char (info++ ? ',' : '(');
-    out_sid(current->sid);
-  }
+#ifdef WITH_SELINUX
   if ( show_scontext ) {
     out_char (info++ ? ',' : '(');
-    out_scontext(current->sid);
+    out_scontext(current->scontext);
   }
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
   if ((swapped && print_args && current->argc < 0) || (!swapped && info))
     out_char (')');
   if (current->highlight && (tmp = tgetstr ("me", NULL)))
     tputs (tmp, 1, putchar);
-#ifdef FLASK_LINUX
-  if (show_scontext || print_args)
-#else  /*FLASK_LINUX*/
   if (print_args)
-#endif /*FLASK_LINUX*/
     {
       for (i = 0; i < current->argc; i++)
 	{
@@ -472,20 +441,20 @@
 	    }
 	}
     }
-#ifdef FLASK_LINUX
+#ifdef WITH_SELINUX
   if ( show_scontext || print_args || ! current->children )
-#else  /*FLASK_LINUX*/
+#else  /*WITH_SELINUX*/
   if (print_args || !current->children)
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
     {
       while (closing--)
 	out_char (']');
       out_newline ();
-#ifdef FLASK_LINUX
+#ifdef WITH_SELINUX
       if ( show_scontext || print_args )
-#else /*FLASK_LINUX*/
+#else /*WITH_SELINUX*/
       if (print_args)
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
 	{
 	  more[level] = !last;
 	  width[level] = swapped + (comm_len > 1 ? 0 : -1);
@@ -575,9 +544,10 @@
   pid_t pid, ppid;
   int fd, size;
   int empty;
-#ifdef FLASK_LINUX
-  security_id_t sid = -1;
-#endif /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+  security_context_t scontext = NULL;
+  int selinux_enabled=(is_selinux_enabled() > 0);
+#endif /*WITH_SELINUX*/
 
   if (!print_args)
     buffer = NULL;
@@ -602,11 +572,15 @@
 	  {
 	    empty = 0;
 	    sprintf (path, "%s/%d", PROC_BASE, pid);
-#ifdef FLASK_LINUX
-            if (fstat_secure(fileno(file),&st,&sid) < 0)
-#else /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+	    if (selinux_enabled)
+	      if (getpidcon(pid,&scontext) < 0)
+		{
+		  perror (path);
+		  exit (1);
+		}
+#endif /*WITH_SELINUX*/
             if (stat (path, &st) < 0)
-#endif /*FLASK_LINUX*/
 	    {
 		perror (path);
 		exit (1);
@@ -631,11 +605,11 @@
 		 &ppid) == 4)
  */
 		if (!print_args)
-#ifdef FLASK_LINUX
-		  add_proc(comm, pid, ppid, st.st_uid, NULL, 0, sid);
-#else  /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+		  add_proc(comm, pid, ppid, st.st_uid, NULL, 0, scontext);
+#else  /*WITH_SELINUX*/
 		  add_proc (comm, pid, ppid, st.st_uid, NULL, 0);
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
 		else
 		  {
 		    sprintf (path, "%s/%d/cmdline", PROC_BASE, pid);
@@ -652,11 +626,11 @@
 		    (void) close (fd);
 		    if (size)
 		      buffer[size++] = 0;
-#ifdef FLASK_LINUX
-		    add_proc(comm, pid, ppid, st.st_uid, buffer, size, sid);
-#else  /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+		    add_proc(comm, pid, ppid, st.st_uid, buffer, size, scontext);
+#else  /*WITH_SELINUX*/
 		    add_proc (comm, pid, ppid, st.st_uid, buffer, size);
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
 		  }
 		}
 	      }
@@ -695,11 +669,11 @@
 	cmd = comm;
       if (*cmd == '-')
 	cmd++;
-#ifdef FLASK_LINUX
-      add_proc(cmd, pid, ppid, uid, NULL, 0, -1);
-#else  /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+      add_proc(cmd, pid, ppid, uid, NULL, 0, NULL);
+#else  /*WITH_SELINUX*/
       add_proc (cmd, pid, ppid, uid, NULL, 0);
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
     }
 }
 
@@ -725,8 +699,7 @@
     "    -u     show uid transitions\n"));
 #ifdef FLASK_LINUX
   fprintf (stderr, _(
-    "    -s     show Flask SIDs\n"
-    "    -x     show Flask security contexts\n"));
+    "    -Z     show SELinux security contexts\n"));
 #endif /*FLASK_LINUX*/
   fprintf (stderr, _(
     "    -U     use UTF-8 (Unicode) line drawing characters\n"
@@ -802,11 +775,11 @@
     sym = &sym_ascii;
   }
 
-#ifdef FLASK_LINUX
-  while ((c = getopt (argc, argv, "aAcGhH:npluUVsx")) != EOF)
-#else  /*FLASK_LINUX*/
+#ifdef WITH_SELINUX
+  while ((c = getopt (argc, argv, "aAcGhH:npluUVZ")) != EOF)
+#else  /*WITH_SELINUX*/
   while ((c = getopt (argc, argv, "aAcGhH:npluUV")) != EOF)
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
     switch (c)
       {
       case 'a':
@@ -862,14 +835,14 @@
       case 'V':
       print_version();
 	return 0;
-#ifdef FLASK_LINUX
-      case 's':
-        show_sids = 1;
-        break;
-      case 'x':
-        show_scontext = 1;
+#ifdef WITH_SELINUX
+      case 'Z':
+	if (is_selinux_enabled() > 0)
+	  show_scontext = 1;
+	else
+	  fprintf(stderr, _("Warning: -Z ignored. Requires an SELinux enabled kernel\n"));
         break;
-#endif /*FLASK_LINUX*/
+#endif /*WITH_SELINUX*/
       default:
 	usage ();
       }
--- psmisc-21.6/configure.in.orig	2005-03-20 22:16:22.860958272 +0100
+++ psmisc-21.6/configure.in	2005-03-20 22:34:25.650349264 +0100
@@ -12,17 +12,17 @@
 AC_PROG_LN_S
 
 dnl checks for options
-AC_SUBST([FLASK_LINUX])
-AC_ARG_ENABLE([flask],
-  [AS_HELP_STRING([--enable-flask], [Enable Security-Enhanced Linux features])],
-  [enable_flask=$enableval],
-  [enable_flask="no"])
-if test "$enable_flask" = "yes"; then
-  AC_DEFINE([FLASK_LINUX], [1], [Use Security-Enhanced Linux features])
-  AC_CHECK_LIB([secure], [avc_toggle], [FLASK_LIB=-lsecure], [
-     AC_MSG_ERROR([Cannot find selinux/ secure static library]) ])
+AC_SUBST([WITH_SELINUX])
+AC_ARG_ENABLE([selinux],
+  [AS_HELP_STRING([--enable-selinux], [Enable Security-Enhanced Linux features])],
+  [enable_selinux=$enableval],
+  [enable_selinux="no"])
+if test "$enable_selinux" = "yes"; then
+  AC_DEFINE([WITH_SELINUX], [1], [Use Security-Enhanced Linux features])
+  AC_CHECK_LIB([selinux], [getfilecon], [SELINUX_LIB=-lselinux], [
+     AC_MSG_ERROR([Cannot find selinux library]) ])
 fi
-AC_SUBST([FLASK_LIB])
+AC_SUBST([SELINUX_LIB])
 
 dnl Checks for libraries.
 AC_CHECK_LIB([tinfo], [tgetent], [TERMCAP_LIB=-ltinfo], [
--- psmisc-21.6/po/pl.po.orig	2005-03-11 11:34:41.000000000 +0100
+++ psmisc-21.6/po/pl.po	2005-03-20 22:42:27.770055928 +0100
@@ -7,7 +7,7 @@
 msgstr ""
 "Project-Id-Version: psmisc 21.6pre1\n"
 "Report-Msgid-Bugs-To: csmall@small.dropbear.id.au\n"
-"POT-Creation-Date: 2005-02-23 14:46+1100\n"
+"POT-Creation-Date: 2005-03-20 22:40+0100\n"
 "PO-Revision-Date: 2005-02-23 20:40+0100\n"
 "Last-Translator: Jakub Bogusz <qboosh@pld-linux.org>\n"
 "Language-Team: Polish <translation-team-pl@lists.sourceforge.net>\n"
@@ -15,32 +15,32 @@
 "Content-Type: text/plain; charset=ISO-8859-2\n"
 "Content-Transfer-Encoding: 8bit\n"
 
-#: src/killall.c:53
+#: src/killall.c:54
 #, c-format
 msgid "Kill %s(%s%d) ? (y/N) "
 msgstr "Zabiæ %s(%s%d)? (t/N) "
 
-#: src/killall.c:245
+#: src/killall.c:241
 #, c-format
 msgid "skipping partial match %s(%d)\n"
 msgstr "pominiêto czê¶ciowe dopasowanie %s(%d)\n"
 
-#: src/killall.c:339
+#: src/killall.c:337
 #, c-format
 msgid "Killed %s(%s%d) with signal %d\n"
 msgstr "Zabito %s(%s%d) sygna³em %d\n"
 
-#: src/killall.c:351 src/fuser.c:494 src/pstree.c:673
+#: src/killall.c:349 src/fuser.c:494 src/pstree.c:647
 #, c-format
 msgid "%s is empty (not mounted ?)\n"
 msgstr "%s jest pusty (nie podmontowany?)\n"
 
-#: src/killall.c:357
+#: src/killall.c:355
 #, c-format
 msgid "%s: no process killed\n"
 msgstr "%s: nie zabito ¿adnego procesu\n"
 
-#: src/killall.c:388
+#: src/killall.c:386
 #, c-format
 msgid ""
 "usage: pidof [ -eg ] name ...\n"
@@ -61,18 +61,17 @@
 "    -V    wy¶wietlenie informacji o wersji\n"
 "\n"
 
-#: src/killall.c:402
+#: src/killall.c:400
 #, c-format
-msgid "usage: killall [-s sid] [-c context] [ -egiqvw ] [ -signal ] name ...\n"
-msgstr ""
-"sk³adnia: killall [-s sid] [-c kontekst] [ -egiqvw ] [ -sygna³ ] nazwa ...\n"
+msgid "usage: killall [-Z context] [ -egiqvw ] [ -signal ] name ...\n"
+msgstr "sk³adnia: killall [-Z kontekst] [ -egiqvw ] [ -sygna³ ] nazwa ...\n"
 
-#: src/killall.c:405
+#: src/killall.c:403
 #, c-format
 msgid "usage: killall [ OPTIONS ] [ -- ] name ...\n"
 msgstr "sk³adnia: killall [ OPCJE ] [ -- ] nazwa ...\n"
 
-#: src/killall.c:408
+#: src/killall.c:406
 #, c-format
 msgid ""
 "       killall -l, --list\n"
@@ -107,20 +106,18 @@
 "  -w,--wait           zaczekanie na ¶mieræ procesu\n"
 "\n"
 
-#: src/killall.c:422
+#: src/killall.c:420
 #, c-format
 msgid ""
-"  -d,--sid            kill only process(es) having sid\n"
-"  -c,--context        kill only process(es) having scontext\n"
-"   (-s, -c are mutually exclusive and must precede other arguments)\n"
+"  -Z,--context        kill only process(es) having scontext\n"
+"   (-Z must precede other arguments)\n"
 "\n"
 msgstr ""
-"  -d,--sid            zabicie tylko procesu(ów) maj±cych dany sid\n"
-"  -c,--context        zabicie tylko procesu(ów) maj±cych dany kontekst "
-"bezp.\n"
-"   (-s i -c wzajemnie siê wykluczaj± i musz± poprzedzaæ inne argumenty)\n"
+"  -Z,--context        zabicie tylko procesu(ów) maj±cych dany kontekst bezp.\n"
+"   (-Z musi poprzedzaæ inne argumenty)\n"
+"\n"
 
-#: src/killall.c:444 src/fuser.c:989 src/pstree.c:743
+#: src/killall.c:441 src/fuser.c:989 src/pstree.c:716
 #, c-format
 msgid ""
 "Copyright (C) 1993-2002 Werner Almesberger and Craig Small\n"
@@ -129,7 +126,7 @@
 "Copyright (C) 1993-2002 Werner Almesberger i Craig Small\n"
 "\n"
 
-#: src/killall.c:446 src/fuser.c:991 src/pstree.c:745
+#: src/killall.c:443 src/fuser.c:991 src/pstree.c:718
 #, c-format
 msgid ""
 "PSmisc comes with ABSOLUTELY NO WARRANTY.\n"
@@ -142,12 +139,12 @@
 "Powszechnej Licencji Publicznej GNU (General Public License).\n"
 "Wiêcej informacji znajduje siê w pliku o nazwie COPYING.\n"
 
-#: src/killall.c:563
+#: src/killall.c:549
 #, c-format
-msgid "%s: SID (%s) must be numeric\n"
-msgstr "%s: SID (%s) musi byæ liczb±\n"
+msgid "Warning: -Z (--context) ignored. Requires an SELinux enabled kernel\n"
+msgstr "Uwaga: -Z (--context) zignorowane. Ta opcja wymaga j±dra z w³±czonym SELinuksem\n"
 
-#: src/killall.c:614
+#: src/killall.c:574
 #, c-format
 msgid "Maximum number of names is %d\n"
 msgstr "Maksymalna liczba nazw to %d\n"
@@ -285,12 +282,12 @@
 msgid "%s/%s: invalid specification\n"
 msgstr "%s/%s: b³êdne okre¶lenie\n"
 
-#: src/pstree.c:377
+#: src/pstree.c:354
 #, c-format
 msgid "MAX_DEPTH not big enough.\n"
 msgstr "MAX_DEPTH jest zbyt ma³e.\n"
 
-#: src/pstree.c:713
+#: src/pstree.c:687
 #, c-format
 msgid ""
 "usage: pstree [ -a ] [ -c ] [ -h | -H pid ] [ -l ] [ -n ] [ -p ] [ -u ]\n"
@@ -323,16 +320,12 @@
 "    -p     wy¶wietlanie PID-ów; w³±cza tak¿e -c\n"
 "    -u     wy¶wietlanie zmian uidów\n"
 
-#: src/pstree.c:728
+#: src/pstree.c:702
 #, c-format
-msgid ""
-"    -s     show Flask SIDs\n"
-"    -x     show Flask security contexts\n"
-msgstr ""
-"    -s     wy¶wietlanie SID-ów Flask\n"
-"    -x     wy¶wietlanie kontekstów bezpieczeñstwa Flask\n"
+msgid "    -Z     show SELinux security contexts\n"
+msgstr "    -Z     wy¶wietlanie kontekstów bezpieczeñstwa SELinuksa\n"
 
-#: src/pstree.c:732
+#: src/pstree.c:705
 #, c-format
 msgid ""
 "    -U     use UTF-8 (Unicode) line drawing characters\n"
@@ -347,32 +340,37 @@
 " u¿ytkownik  tylko drzewa zaczynaj±ce siê od procesów tego u¿ytkownika\n"
 "\n"
 
-#: src/pstree.c:741
+#: src/pstree.c:714
 #, c-format
 msgid "pstree (psmisc) %s\n"
 msgstr "pstree (psmisc) %s\n"
 
-#: src/pstree.c:835
+#: src/pstree.c:809
 #, c-format
 msgid "TERM is not set\n"
 msgstr "TERM nie ustawiony\n"
 
-#: src/pstree.c:840
+#: src/pstree.c:814
 #, c-format
 msgid "Can't get terminal capabilities\n"
 msgstr "Nie mo¿na odczytaæ mo¿liwo¶ci terminala\n"
 
-#: src/pstree.c:884
+#: src/pstree.c:844
+#, c-format
+msgid "Warning: -Z ignored. Requires an SELinux enabled kernel\n"
+msgstr "Uwaga: -Z zignorowane. Ta opcja wymaga j±dra z w³±czonym SELinuksem\n"
+
+#: src/pstree.c:858
 #, c-format
 msgid "No such user name: %s\n"
 msgstr "Nie ma u¿ytkownika o takiej nazwie: %s\n"
 
-#: src/pstree.c:900
+#: src/pstree.c:874
 #, c-format
 msgid "No processes found.\n"
 msgstr "Nie znaleziono procesów.\n"
 
-#: src/pstree.c:905
+#: src/pstree.c:879
 #, c-format
 msgid "Press return to close\n"
 msgstr "Naci¶niêcie return zamknie program\n"
