--- psmisc-22.11/configure.ac~	2010-04-09 23:22:57.393791668 +0200
+++ psmisc-22.11/configure.ac	2010-04-09 23:28:01.826051280 +0200
@@ -65,7 +65,10 @@
 		struct user_regs_struct.rax,
 		struct user_regs_struct.rdi,
 		struct user_regs_struct.rsi,
-		struct user_regs_struct.rdx], [],[], [#include <sys/user.h>])
+		struct user_regs_struct.rdx], [],[], [
+					       #include <sys/types.h>
+					       #include <sys/user.h>
+					       ])
 AC_CHECK_MEMBERS([struct pt_regs.orig_gpr3,
 		struct pt_regs.gpr], [],[], [#include <linux/ptrace.h>])
 AM_CONDITIONAL(WANT_PEEKFD_I386,
