diff -urN psmisc-20.1.org/configure.in psmisc-20.1/configure.in
--- psmisc-20.1.org/configure.in	Sat Mar 31 18:45:47 2001
+++ psmisc-20.1/configure.in	Sat Mar 31 18:46:30 2001
@@ -8,9 +8,10 @@
 AC_PROG_CPP
 
 dnl Checks for libraries.
-AC_CHECK_LIB(ncurses, tgetent, TERMCAP_LIB=-lncurses,
-  AC_CHECK_LIB(termcap, tgetent, TERMCAP_LIB=-ltermcap, 
-    AC_MSG_ERROR(Cannot find ncurses or termcap libraries) ) )
+AC_CHECK_LIB(tinfo, tgetent, TERMCAP_LIB=-ltinfo,
+  AC_CHECK_LIB(ncurses, tgetent, TERMCAP_LIB=-lncurses,
+    AC_CHECK_LIB(termcap, tgetent, TERMCAP_LIB=-ltermcap, 
+      AC_MSG_ERROR(Cannot find ncurses or termcap libraries) ) ) )
 AC_SUBST(TERMCAP_LIB)
 
 dnl Checks for header files.
diff -urN psmisc-20.1.org/doc/Makefile.am psmisc-20.1/doc/Makefile.am
--- psmisc-20.1.org/doc/Makefile.am	Sat Mar 31 18:45:47 2001
+++ psmisc-20.1/doc/Makefile.am	Sat Mar 31 19:01:08 2001
@@ -1,4 +1,4 @@
 
-man_MANS = fuser.1 killall.1 pidof.1 pstree.1
+man_MANS = fuser.1 killall.1 pstree.1
 
 EXTRA_DIST = $(man_MANS)
--- psmisc-20.2/src/Makefile.am.orig	Fri Jun 15 14:17:55 2001
+++ psmisc-20.2/src/Makefile.am	Sun Oct 21 02:56:54 2001
@@ -1,7 +1,8 @@
 
-CFLAGS = -Wall -DLOCALEDIR=\"/usr/share/locale\"
+DEFS = @DEFS@ -DLOCALEDIR=\"/usr/share/locale\"
 
-bin_PROGRAMS = fuser killall pstree
+bin_PROGRAMS = fuser
+sbin_PROGRAMS = killall pstree
 
 fuser_SOURCES = fuser.c comm.h signals.c signals.h signames.h loop.h
 
@@ -9,7 +10,7 @@
 
 pstree_SOURCES = pstree.c comm.h
 
-pstree_LDADD = @TERMCAP_LIB@
+pstree_LDADD = $(TERMCAP_LIB)
 
 BUILT_SOURCES = signames.h
 
@@ -24,5 +25,3 @@
 		  { rm -f signames.h; exit 1; }
 		grep '^{ 1,"HUP" },$$' signames.h >/dev/null || \
 		  { rm -f signames.h; exit 1; }
-
-
