--- psmisc-21.6/configure.in.orig	2005-03-11 11:57:41.000000000 +0100
+++ psmisc-21.6/configure.in	2005-03-20 22:13:40.404655384 +0100
@@ -25,9 +25,10 @@
 AC_SUBST([FLASK_LIB])
 
 dnl Checks for libraries.
+AC_CHECK_LIB([tinfo], [tgetent], [TERMCAP_LIB=-ltinfo], [
 AC_CHECK_LIB([ncurses], [tgetent], [TERMCAP_LIB=-lncurses], [
   AC_CHECK_LIB([termcap], [tgetent], [TERMCAP_LIB=-ltermcap], [
-    AC_MSG_ERROR([Cannot find ncurses or termcap libraries]) ]) ])
+    AC_MSG_ERROR([Cannot find ncurses or termcap libraries]) ]) ]) ])
 AC_SUBST([TERMCAP_LIB])
 
 dnl Checks for header files.
